src/engine/SCons/CacheDir.py

```diff
@@ -159,14 +221,15 @@ def __init__(self, path):
             # as an attempt to alleviate this, on the basis that it's a pretty
             # unlikely occurence (it'd require two builds with a brand new cache
             # directory)
-            if os.path.isdir(path) and len(os.listdir(path)) != 0:
+            #if os.path.isdir(path) and len(os.listdir(path)) != 0:
+            if os.path.isdir(path) and len([f for f in os.listdir(path) if os.path.basename(f) != "config"]) != 0:
```

Comment:
Can you get rid of above line, and maybe use Any(os.listdir(path)) ?

Reply:
Sure, I can look at that. I just clipped this one from the issue (#3351) so some improvement could happen in the py2.7 case as well.